cmd/swarm/swarm-smoke/upload_and_sync.go

```diff
@@ -65,58 +63,93 @@ func uploadAndSyncCmd(ctx *cli.Context, tuid string) error {
 	}
 
 	// trigger debug functionality on randomBytes
-	e := trackChunks(randomBytes[:])
+	e := trackChunks(randomBytes[:], true)
 	if e != nil {
 		log.Error(e.Error())
 	}
 
 	return err
 }
 
-func trackChunks(testData []byte) error {
+func trackChunks(testData []byte, submitMetrics bool) error {
 	addrs, err := getAllRefs(testData)
 	if err != nil {
+		log.Error("cannot get refs", "err", err.Error())
 		return err
 	}
 
 	for i, ref := range addrs {
-		log.Trace(fmt.Sprintf("ref %d", i), "ref", ref)
+		log.Debug(fmt.Sprintf("ref %d", i), "ref", ref)
 	}
 
+	var globalYes, globalNo int
+	var globalMu sync.Mutex
+	var hasErr bool
+
+	var wg sync.WaitGroup
+	wg.Add(len(hosts))
+
 	for _, host := range hosts {
-		httpHost := fmt.Sprintf("ws://%s:%d", host, 8546)
+		host := host
+		go func() {
+			defer wg.Done()
+			httpHost := fmt.Sprintf("ws://%s:%d", host, 8546)
+
+			ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+			defer cancel()
+			rpcClient, err := rpc.DialContext(ctx, httpHost)
+			if rpcClient != nil {
+				defer rpcClient.Close()
+			}
+			if err != nil {
+				log.Error("error dialing host", "err", err, "host", httpHost)
+				hasErr = true
+				return
+			}
 
-		hostChunks := []string{}
+			var hostChunks string
+			err = rpcClient.Call(&hostChunks, "bzz_has", addrs)
+			if err != nil {
+				log.Error("error calling rpc client", "err", err, "host", httpHost)
+				hasErr = true
+				return
+			}
 
-		rpcClient, err := rpc.Dial(httpHost)
-		if err != nil {
-			log.Error("error dialing host", "err", err, "host", httpHost)
-			continue
-		}
+			yes, no := 0, 0
+			for _, val := range hostChunks {
+				if val == '1' {
+					yes++
+				} else {
+					no++
+				}
+			}
 
-		var hasInfo []api.HasInfo
-		err = rpcClient.Call(&hasInfo, "bzz_has", addrs)
-		if err != nil {
-			log.Error("error calling rpc client", "err", err, "host", httpHost)
-			continue
-		}
+			if no == 0 {
```

Comment:
this doesn't cover `yes==0` (if returned array was empty for some reason)

Reply:
We don't need to check for `yes==0`. Even this check is not necessary, as it is only for a log line, but it makes it easy to browse the logs.